controllers/pgbench/job.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2019 The xridge kubestone contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pgbench
+
+import (
+	"fmt"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/firepear/qsplit"
+	perfv1alpha1 "github.com/xridge/kubestone/api/v1alpha1"
+	"github.com/xridge/kubestone/controllers/common"
+)
+
+// NewJob creates a new pgbench job
+func NewJob(cr *perfv1alpha1.Pgbench) *batchv1.Job {
+	objectMeta := metav1.ObjectMeta{
+		Name:      cr.Name,
+		Namespace: cr.Namespace,
+	}
+
+	env := []corev1.EnvVar{
+		{Name: "PGHOST", Value: cr.Spec.Postgres.Host},
+		{Name: "PGPORT", Value: fmt.Sprintf("%d", cr.Spec.Postgres.Port)},
+		{Name: "PGUSER", Value: cr.Spec.Postgres.User},
+		{Name: "PGPASSWORD", Value: cr.Spec.Postgres.Password},
+		{Name: "PGDATABASE", Value: cr.Spec.Postgres.Database},
+	}
+
+	initContainer := corev1.Container{
```

Comment:
I'm not sure if init container is the right way to solve this problem.
If you consider execution of multiple benchmarks, only the first one needs to initialize the DB.

I also miss the teardown part. 

Would it make sense to have the following flags in the CR:
 - initEnabled
 - benchmarkEnabled
 - teardownEnabled

Having these would enable the user to execute init exactly once, without a need of a benchmark,
run multiple benchmarks sequentially / parallelly and also clean up at the end. WDYT?

Reply:
> If you consider execution of multiple benchmarks, only the first one needs to initialize the DB.
> I also miss the teardown part.

I couldn't find any example for a teardown for pgbench. However, when running pgbench for the second time, it handles dropping the previously generated tables. Based on this, I'm not sure a teardown is needed here.

Whatever your opinion is about this, can we agree on doing any flow related modification in a follow up PR?

Reply:
yep, let's do that.